src/routes/auth/guards/auth.guard.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  Injectable,
+} from '@nestjs/common';
+import { IAuthRepository } from '@/domain/auth/auth.repository.interface';
+import { AuthService } from '@/routes/auth/auth.service';
+import { SiweMessageSchema } from '@/domain/auth/entities/siwe-message.entity';
+
+/**
+ * The AuthGuard should be used to protect routes that require authentication.
+ *
+ * It checks for the presence of a valid JWT access token in the request and
+ * verifies its validity before allowing access to the route.
+ *
+ * 1. Check for the presence of an access token in the request.
+ * 2. Verify the token's validity and decode it.
+ * 3. Ensure the decoded token is that of a valid SiweMessage.
+ * 4. Ensure the token has not expired and is already valid.
+ * 5. Allow access if all checks pass.
+ */
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  constructor(
+    @Inject(IAuthRepository) private readonly authRepository: IAuthRepository,
+  ) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+
+    const accessToken = this.authRepository.getAccessToken(
+      request,
+      AuthService.AUTH_TOKEN_TOKEN_TYPE,
+    );
+
+    // No token in the request
+    if (!accessToken) {
+      return false;
+    }
+
+    const payload = this.authRepository.verifyAccessToken(accessToken);
```

Comment:
The schema of a JWT payload might not match the one required by SIWE right? In JWT the payload would contain a set of claims which is specific to JWT – https://www.iana.org/assignments/jwt/jwt.xhtml

Reply:
As of `0d0c355763a73da4bb910f1fd82f33b90bcef21e`, we are now only storing `signer_address` in the payload.